packages/dialect-react-sdk/hooks/useUnreadMessages.ts

```diff
@@ -1,37 +1,77 @@
 import type { PublicKey } from '@solana/web3.js';
-import useSWR from 'swr';
-import { CACHE_KEY_THREAD_SUMMARY_FN } from './internal/swrCache';
+import useSWR, { KeyedMutator } from 'swr';
+import {
+  CACHE_KEY_THREAD_SUMMARY_FN,
+  CACHE_KEY_THREADS_SUMMARY,
+} from './internal/swrCache';
 import useDialectSdk from './useDialectSdk';
+import type {
+  ThreadsGeneralSummary,
+  ThreadSummary,
+} from '../../../../sdk/packages/sdk';
+import { EMPTY_OBJ } from '../utils';
 
-interface UseUnreadMessagesParams {
-  otherMembers: PublicKey[];
+interface BaseUseUnreadMessagesParams {
   refreshInterval?: number;
 }
+interface PerThreadUseUnreadMessagesParams extends BaseUseUnreadMessagesParams {
+  otherMembers: PublicKey[];
+}
 
-interface UseUnreadMessagesValue {
+interface UseUnreadMessageValue {
+  unreadCount: number;
+  refresh: () => void;
   hasUnreadMessages: boolean;
 }
 
-const useUnreadMessages = ({
-  otherMembers,
-  refreshInterval,
-}: UseUnreadMessagesParams): UseUnreadMessagesValue => {
+const isParamsExtended = (
+  params: BaseUseUnreadMessagesParams | PerThreadUseUnreadMessagesParams
+): params is PerThreadUseUnreadMessagesParams =>
+  !!(params as PerThreadUseUnreadMessagesParams).otherMembers;
```

Comment:
very minor, but I'd use `Boolean()` instead of `!!`

Reply:
K, done, sir!